### PR TITLE
ci(gh-pages): Log GitHub Pages deployment errors

### PR DESCRIPTION
**Pull request by markdalgleish**
Deployments are still failing for some reason, but it's not reflected in the build logs. I realised I forgot to log deployment errors, so I've added some extra logging, but a refactor was required to ensure that all console logs are filtered.

### CODE REVIEW

File: scripts/deploy-gh-pages.js [hunk 1]
```diff
@@ -3,27 +3,38 @@ var ghpages = require('gh-pages');
 var basePath = path.join(__dirname, '../dist');
 var repoUrl = require('../package.json').repository.url;
 
-var makeConfig = function(token) {
-  if (token) {
+var GH_TOKEN = process.env.GH_TOKEN;
+var tokenRegex = GH_TOKEN ? new RegExp(GH_TOKEN, 'g') : null;
```
Comment (jahredhope) [inline scripts/deploy-gh-pages.js]: Does the GH_TOKEN ever contain Regex special characters?

File: scripts/deploy-gh-pages.js [hunk 2]
```diff
@@ -3,27 +3,38 @@ var ghpages = require('gh-pages');
...
+var log = function(message) {
+  if (tokenRegex) {
+    // Hide token from build logs
+    message = message.replace(tokenRegex, '[GH_TOKEN]');
+  }
+
+  console.log(message);
+};
+
+var makeConfig = function() {
```
Comment (mjt01) [inline scripts/deploy-gh-pages.js]: This function could be simplified now:
```
return {
  repo: GH_TOKEN ? repoUrl.replace(/^https?:\/\//, 'http://' + GH_TOKEN + '@') : repoUrl,
  logger: log
};
```

File: scripts/deploy-gh-pages.js (hunk 1 above)
Comment (markdalgleish) [inline scripts/deploy-gh-pages.js]: It shouldn't, it's just alphanumeric.

File: scripts/deploy-gh-pages.js (hunk 2 above)
Comment (markdalgleish) [inline scripts/deploy-gh-pages.js]: I've been deliberately optimising for readable config without forcing you to execute the code in your head, hence why I'm maintaining two separate objects. Does that sound okay?

Comment (mjt01) [inline scripts/deploy-gh-pages.js]: I hear what you're saying but not sure it's clearer having nested if statements that all hinge on the presence of `GH_TOKEN`.

**Pull request merged**